hedera-node/hedera-addressbook-service-impl/src/main/java/com/hedera/node/app/service/addressbook/impl/handlers/NodeGetInfoHandler.java

```diff
@@ -74,7 +87,50 @@ public Response findResponse(@NonNull final QueryContext context, @NonNull final
         requireNonNull(header);
         final var query = context.query();
         final var config = context.configuration().getConfigData(LedgerConfig.class);
-        throw new UnsupportedOperationException("need implementation");
+        final var nodeStore = context.createStore(ReadableNodeStore.class);
+        final var op = query.nodeGetInfoOrThrow();
+        final var response = NodeGetInfoResponse.newBuilder();
+        final var node = NodeInfo.newBuilder().nodeId(op.nodeId()).build();
+        response.nodeInfo(node);
+
+        final var responseType = op.headerOrElse(QueryHeader.DEFAULT).responseType();
+        response.header(header);
+        if (header.nodeTransactionPrecheckCode() == OK && responseType != COST_ANSWER) {
+            final var optionalInfo = infoForNode(op.nodeId(), nodeStore, config);
+            optionalInfo.ifPresent(response::nodeInfo);
+        }
+
+        return Response.newBuilder().nodeGetInfo(response).build();
+    }
+
+    /**
+     * Provides information about a node.
+     * @param nodeId the node to get information about
+     * @param nodeStore the node store
+     * @param config the LedgerConfig
+     * @return the information about the node
+     */
+    private Optional<NodeInfo> infoForNode(
+            @NonNull final long nodeId,
+            @NonNull final ReadableNodeStore nodeStore,
+            @NonNull final LedgerConfig config) {
+        final var meta = nodeStore.get(nodeId);
+        if (meta == null) {
+            return Optional.empty();
+        } else {
+            final var info = NodeInfo.newBuilder();
+            info.nodeId(meta.nodeId());
+            if (meta.accountId() != null) info.accountId(meta.accountId());
```

Comment:
We can remove if check here. @Nullable AccountID accountId for NodeInfo, although we checked node.accountId won't be null during create and update.

Reply:
Removed thank you.